Sources/Datadog/DatadogCore/DatadogCore.swift

```diff
@@ -97,6 +97,21 @@ internal final class DatadogCore {
         )
     }
 
+    /// Add or override the properties of the current user
+    ///
+    ///  - Parameters:
+    ///    - parameters: The user's custom attibutes to add over override
+    func addUserProperties(properties: [AttributeKey: AttributeValue?]) {
+        var extraInfo = dependencies.userInfoProvider.value.extraInfo
+        for property in properties {
```

Comment:
There is much cleaner (and faster) way in swift via `compactMap` than `for` cycle (and you are kind of using it below, not sure why not using here too):

```swift
extension Dictionary {
   func removeOptionalValues<T>() -> [Key: T] where Value == T? {
      return compactMapValues { $0 }
   }
}
```
You could name it `removeNilValues` or `removeOptionalValues` or anything :)

Reply:
This isn't removing nil values in extraInfo. It's removing values from extraInfo who's values are `nil` in the properties map.

Reply:
My bad, sorry! :)